internal/pkg/opentelemetry/controller_test.go

```diff
@@ -0,0 +1,239 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentelemetry
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-logr/stdr"
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.25.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"go.opentelemetry.io/auto/internal/pkg/instrumentation/probe"
+)
+
+// copied from instrumentation.go.
+func instResource() *resource.Resource {
```

Comment:
Is there a way to avoid the duplication? Mabye move the `res()` function to another location?

Reply:
the `res()` function is pretty buried within the instrumentation.go code, so I think it would take a bigger refactor of that code to pull it out in any easily re-usable way.

Besides that, this code isn't testing that function anyway. This copy doesn't include all of the `instConfig.res()` code, because this test doesn't need to rely on build info or any of the other fields from an instConfig. So pulling it directly in here isn't relevant to these tests